src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -944,6 +944,22 @@ public function fresh($with = [])
     }
 
     /**
+     * Reload the current model instance with fresh attributes from the database.
+     *
+     * @return void
+     */
+    public function refresh()
+    {
+        if (! $this->exists) {
+            return;
+        }
+
+        $this->load(array_keys($this->relations));
+
+        $this->setRawAttributes(static::find($this->getKey())->attributes);
```

Comment:
Should this be `findOrFail`? What should happen if the model has been deleted in the meantime?